fuel-core/src/tx_pool.rs

```diff
@@ -83,9 +85,50 @@ impl TxPool {
 
     pub async fn submit_tx(&self, tx: Transaction) -> Result<Bytes32, Error> {
         let db = self.db.clone();
-        let tx_id = tx.id();
+
+        let mut tx_to_exec = tx.clone();
+
+        let includable_txs: Vec<Transaction>;
+
+        if self.executor.config.utxo_validation {
+            if tx_to_exec.metadata().is_none() {
+                tx_to_exec.precompute_metadata();
+            }
+
+            let results = self
+                .fuel_txpool
+                .insert(vec![Arc::new(tx_to_exec.clone())])
+                .await;
+
+            // Error handling a WIP, just saving for now
+            if results.get(0).unwrap().is_err() {
+                // Return error if tx failed to insert
+                let failed_with = results.get(0).unwrap().as_ref().err().unwrap().to_string();
+
+                return Err(Error::String(failed_with));
+            }
```

Comment:
```suggestion
            self.fuel_txpool
                .insert(vec![Arc::new(tx_to_exec.clone())])
                .await
                .into_iter()
                .collect::<Result<Vec<_>, _>>()?;
```

Reply:
https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect

Reply:
Didn't know collect() could also handle errors, will start trying to look out for areas where using `iter()` can improve the elegance of my code, thanks for the doc page, that is helpful in explaining it